test/sampleapps/sampleapps.go

```diff
@@ -18,28 +18,30 @@ const (
 	Namespace = "test-namespace-1"
 )
 
-func Restart(ctx context.Context, t *testing.T, config *envconf.Config) context.Context {
-	var sampleDeployment appsv1.Deployment
-	var pods corev1.PodList
-	resource := config.Client().Resources()
+func Restart(namespaceName string) func(ctx context.Context, t *testing.T, config *envconf.Config) context.Context {
+	return func(ctx context.Context, t *testing.T, config *envconf.Config) context.Context {
+		var sampleDeployment appsv1.Deployment
+		var pods corev1.PodList
+		resource := config.Client().Resources()
 
-	require.NoError(t, resource.WithNamespace(Namespace).List(ctx, &pods))
+		require.NoError(t, resource.WithNamespace(namespaceName).List(ctx, &pods))
 
-	for _, podItem := range pods.Items {
-		require.NoError(t, resource.Delete(ctx, &podItem))
-	}
+		for _, podItem := range pods.Items {
+			require.NoError(t, resource.Delete(ctx, &podItem))
+		}
 
-	require.NoError(t, resource.Get(ctx, Name, Namespace, &sampleDeployment))
-	require.NoError(t, wait.For(
-		conditions.New(resource).DeploymentConditionMatch(
-			&sampleDeployment, appsv1.DeploymentAvailable, corev1.ConditionTrue)))
+		require.NoError(t, resource.Get(ctx, Name, namespaceName, &sampleDeployment))
+		require.NoError(t, wait.For(
+			conditions.New(resource).DeploymentConditionMatch(
+				&sampleDeployment, appsv1.DeploymentAvailable, corev1.ConditionTrue)))
 
-	return ctx
+		return ctx
+	}
 }
 
-func Get(t *testing.T, ctx context.Context, resource *resources.Resources) corev1.PodList {
+func Get(t *testing.T, ctx context.Context, resource *resources.Resources, namespaceName string) corev1.PodList {
```

Comment:
Should we move that to the kubeobjects pod package? - To stay more consistent with other objects e.g. daemonsets and deployments. There we already have a `get.go`

Reply:
sampleapps.Get renamed to pod.List